format/defs.bzl

```diff
@@ -55,30 +63,50 @@ def _format_attr_factory(target_name, lang, toolname, tool_label, mode):
         "data": [tool_label],
     }
 
+languages = rule(
+    implementation = lambda ctx: fail("languages rule is documentation-only; do not call it"),
+    doc = """\
+Language attributes that may be passed to [format_multirun](#format_multirun) or [format_test](#format_test).
+    
+Files with matching extensions from [GitHub Linguist] will be formatted for the given language.
+
+Some languages have dialects:
+    - `javascript` includes TypeScript, TSX, and JSON.
+    - `css` includes Less and Sass.
+
+**Do not call the `languages` rule directly, it exists only to document the attributes.**
+
+[GitHub Linguist]: https://github.com/github-linguist/linguist/blob/559a6426942abcae16b6d6b328147476432bf6cb/lib/linguist/languages.yml
+""",
+    attrs = {
+        to_attribute_name(key): attr.label(
+            doc = "a `{0}` binary, or any other tool that has a matching command-line interface. {1}".format(
+                value,
+                "Use `@aspect_rules_lint//format:{}` to choose the built-in tool.".format(BUILTIN_TOOL_LABELS[key].split("/")[-1]) if key in BUILTIN_TOOL_LABELS.keys() else "",
+            ),
+        )
+        for key, value in TOOLS.items()
+    },
+)
+
 def format_multirun(name, jobs = 4, print_command = False, **kwargs):
-    """Create a multirun binary for the given formatters.
+    """Create a [multirun] binary for the given languages.
 
     Intended to be used with `bazel run` to update source files in-place.
-    To check formatting with `bazel test`, see [format_test](#format_test).
 
-    Also produces a target `[name].check` which does not edit files, rather it exits non-zero
-    if any sources require formatting.
+    This macro produces a target named `[name].check` which does not edit files,
+    rather it exits non-zero if any sources require formatting.
 
-    Tools are provided by default for some languages.
-    These come from the `@multitool` repo.
-    Under --enable_bzlmod, rules_lint creates this automatically.
-    WORKSPACE users will have to set this up manually. See the release install snippet for an example.
+    Not recommended: to check formatting with `bazel test`, use [format_test](#format_test) instead.
```

Comment:
does this mean to say that `[name].check` is not recommended, or that `format_test` is not recommended? Its pretty ambiguous

Reply:
thanks, I'll clarify it's the latter